### ISSUE TITLE
Graph: Add quick filter input field for Legend values in widget

### ISSUE THREAD

**Issue by gelicia** [grafana/grafana#83021]
### Discussed in https://github.com/grafana/grafana/discussions/38213

<div type='discussions-op-text'>

<sup>Originally posted by **MurzNN** January 28, 2021</sup>
**What would you like to be added**:
Will be good to add small input field to type keywords for quick filter-out the whole list of Legend values by containing it's keyword. It can be located in legend name header of Table style, where now is empty place (when enabled Values options).

And maybe similar filter for non-table lists of values. For not waste space in regular charts with only several Legend values, we can show the filter input on-demand, only if chart have scroll line in legend area (or maybe on hover).

**Why is this needed**:
Some charts have very long list of Legend values, and it is too hard to find needed ones. Via quick filter we will can filter-out them by some keyword, here is example of problem and proposed solution:
![image](https://user-images.githubusercontent.com/336662/106111181-ca121400-615c-11eb-8f55-9407f51ceaa8.png)

</div>

**Comment by zffocussss** [comment]
yes,it is very convenient for users to focus on the interest item.
